src/utils/device_permissions.cljs

```diff
@@ -0,0 +1,18 @@
+(ns utils.device-permissions
```

Comment:
I don't think this `camera` function should be defined in a utility namespace. This looks like a simple event handler that should be moved to some `events.cljs` file.

Reply:
@ilmotta this is to have all device permissions handling logic in one file to avoid duplications. For example accessing the camera would be needed in-chat, when editing profile, when editing group, when editing community, and maybe other places too. I don't mind moving it to another namespace if you think `utils` is not the most suitable.

Reply:
Yep seems like code that should be somewhere in status_im2 to me too

Reply:
I understand the reason why you put it here @OmarBasem, but the thing is that this function is an event handler, it dispatches `:request-permissions` and then a toast event. If it needs to be reused in other places, then it's just a matter of defining the event somewhere else and let any view dispatch it, something like `(rf/dispatch [:permissions/camera args])`.

There's no reason to leak re-frame into utility namespaces :)

Reply:
btw, not sure if related but we have this file: src/react-native/permissions.cljs. You're aware of it @OmarBasem ? 🤔

Reply:
@J-Son89 that file is to expose the permissions library in ClojureScript. I think we shouldn't have any re-frame in the `react-native` namespace.

Reply:
Yep for sure. Just was not sure if it was related here/ you were aware of it. Seems like you have it figured out anyway 👌